src/store/tokens.ts

```diff
@@ -79,7 +78,10 @@ export const actions: ActionTree<TokensModuleState, RootState> = {
   async loadAllTokens({ commit, dispatch, getters }): Promise<Tokens> {
     if (Object.entries(getters.getAllTokens).length === 0) {
       await this.dispatch("wallet/restoreProviderConnection");
-      const tokensList = await walletData.get().syncProvider?.getTokens();
+      /* By taking token list from syncProvider we avoid double getTokens request,
+          but the tokensBySymbol param is private on zksync utils types */
+      // @ts-ignore
+      const tokensList = walletData.get().syncProvider!.tokenSet.tokensBySymbol;
```

Comment:
The `tokensBySymbol` was made private so that the clients use the methods like `resolveTokenId`, `resolveTokenSymbol`, etc.

This is definitely out of scope of this PR, but why is it so important to store the object itself? It is a quite risky thing, since at any time the TS transpiler may decide to transpile this field as truly private one, which can not be accessed simply by its name

Reply:
This is used to get the list of tokens, not a single one. When creating zksync provider it requests the list of tokens and then there is no way to receive it from zksync provider cache. It is only possible to create new request, which is obviously not good for the request amount optimization. I actually told you about this like a month ago when we were working on the wallet optimization.

Reply:
Ok. We will fix that in the subsequent zksync.js versions

Reply:
BTW, @JackHamer09 why do we need the list of all tokens?

Reply:
To have information about the tokens. Yeah, we don't need all of them, but if we already have a cached version of all, why wouldn't we use it. And well, another reason is that this part of architecture taken from the wallet.